File: ios/Khrysalis Template/com/lightningkite/khrysalistemplate/vg/SegmentedControlDemoVG.shared.swift
//Package: com.lightningkite.khrysalistemplate.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps
import KhrysalisBluetooth



public class SegmentedControlDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Segmented Control Demo"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = SegmentedControlDemoXml()
        var view = xml.setup(dependency)
        xml.tabs.bind(["A", "B", "C"], StandardObservableProperty(0))
        xml.tabs2.bind(["A", "B", "C", "D", "E", "F", "G"], StandardObservableProperty(0))
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        super.init()
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/vg/SegmentedControlDemoVG.shared.kt
package com.lightningkite.khrysalistemplate.vg

import android.view.View
import com.lightningkite.khrysalis.observables.binding.bind
import com.lightningkite.khrysalis.observables.StandardObservableProperty
import com.lightningkite.khrysalis.views.ViewDependency
import com.lightningkite.khrysalis.views.ViewGenerator
import com.lightningkite.khrysalistemplate.layouts.SegmentedControlDemoXml

class SegmentedControlDemoVG() : ViewGenerator() {
    override val title: String get() = "Segmented Control Demo"

    override fun generate(dependency: ViewDependency): View {
        val xml = SegmentedControlDemoXml()
        val view = xml.setup(dependency)

        xml.tabs.bind(listOf("A", "B", "C"), StandardObservableProperty(0))
        xml.tabs2.bind(listOf("A", "B", "C", "D", "E", "F", "G"), StandardObservableProperty(0))

        return view
    }
}
